packages/peregrine/lib/talons/Wishlist/WishlistButton/useWishlistButton.ce.js

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useMutation } from '@apollo/client';
+import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
+
+import DEFAULT_OPERATIONS from './wishlistButton.gql';
+
+export const useWishlistButton = props => {
+    const { itemOptions } = props;
+    const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
+
+    const { addProductToWishlistMutation } = operations;
+
+    const [isItemAdded, setIsItemAdded] = useState(false);
+
+    useEffect(() => {
+        // If a user changes selections, let them add that combination to a list.
+        if (itemOptions.selected_options) setIsItemAdded(false);
```

Comment:
Or do I go through `handleClick` when that happens?

Reply:
Each option selection is treated as a new item to be added because a) there is no limit to the quantity of a given config you can add and b) there is no memory of whether the item, in its current configuration, was added to the list.

This workflow was approved by UX :)